### PR TITLE
Add composite action to create upstream tag

### PR DESCRIPTION
**Pull request by ochosi**
The release_bot.py contains several steps, which were previously part of
our release script in osbuild/maintainer-tools. The script here is much
shorter though.
Things that were dropped were the sanity checks, because we can presume
that the git repository at hand is sane, that the git remote is well
configured etc.
Furthermore all interactivity of the original script was removed. Even
though I had recently added a --no-interactive mode to the original
script, I just felt this would make the code harder to read.

The rationale for making this a composite action instead of a standalone
repository was that this way each component can define it's own
automated schedule and that there would be some representation in the
GitHub Actions for the component of the tag being created.

I'm happy to drop the Contributors list and the date/location line. Those felt more
meaningful when it was still a human doing the release.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -1,57 +1,44 @@
...
+description: "Create a tag and push it"
 
 inputs:
   token:
-    description: "A GitHub token for creating a release"
+    description: "A GitHub token for creating a tag"
     required: true
-  slack_webhook_url:
-    description: "A Slack incoming Webhook URL"
+  component:
```
Comment (ondrejbudai) [inline action.yml:8]: Cannot we extract this somehow from action's context? Might be way more difficult than it sounds though.

Comment (ondrejbudai) [reply action.yml:8]: https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

`github.repository` maybe?

**Review by ondrejbudai — APPROVED** [review]
Looks good!

File: create_tag.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+
+def get_pullrequest_infos(args, repo, hashes):
+    """Fetch the titles of all related pull requests"""
+    logging.debug("Collect pull request titles...")
+    api = GhApi(repo=repo, owner='ochosi', token=args.token)
+    summaries = []
+    i = 0
+
+    for commit_hash in hashes:
```
Comment (ondrejbudai) [inline create_tag.py:100]:
```suggestion
    for i, commit_hash in enumerate(hashes):
```

is slightly more pythonic.

File: action.yml (hunk 1 above)
Comment (ochosi) [reply action.yml:8]: Yes, thanks. I've thought about this before and even looked in the docs, but couldn't find it. I'll try to test if this is "it".

**Review by ondrejbudai — APPROVED** [review]

**Pull request merged**